lib/elixir/lib/kernel.ex

```diff
@@ -1614,6 +1615,25 @@ defmodule Kernel do
     {:::, [], [other, {:binary, [], nil}]}
   end
 
+  defp check_invalid_argument({:<<>>, _, _}, _caller) do
+    :ok
+  end
+
+  defp check_invalid_argument(arg, %{context: :match} = caller) when not is_binary(arg) do
+    :erlang.error(
```

Comment:
It is fine to raise an argument error. It is a macro and macro have arguments. :)